### PR TITLE
Test Excersice 01

### PR DESCRIPTION
**Pull request by Poetickz**
This PR:
- Creates Aritmetic_Operation class.
- Add passing spec for Aritmetic_Operation class.

### CODE REVIEW

File: ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'rspec/autorun'
+
+class Aritmetic_operations
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: Class names should be CamelCase... In this case, `ArithmeticOperations`. Also, please watch the spelling (missing the "h" in arithmetic).

Comment (HectorMg) [reply ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: Also, the file name should match the class name in snake_case (`arithmetic_operations.rb`) when writing a single class per file.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def self.multiply(first_number, second_number)
+        first_number * second_number
+    end
+
+    def self.divide(first_number, second_number)
+        first_number / second_number
+    end
+end
+
+calculus = Aritmetic_operations.add(12, 4)
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: Should this script code really be in this class file? When writing application code, avoid committing "useless" code such as small script code used to manually test your implementation.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+calculus = Aritmetic_operations.substraction(15,5)
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: Remember adding a space after commas that separate arguments!! (In this case you should remove this code, it's more of a reminder for the future and other cases).

File: ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require 'rspec/autorun'
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: You aren't using any methods or classes from this gem in this file, this line can (should) be removed.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+class Aritmetic_operations
+    
+    def self.add(first_number, second_number)
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/math_maddness.rb]: The variable names are more descriptive than something like "num1" and "num2", but they're still to vague (what does the "number" represent?). In a calculator context, "numbers" are called/referred to as "operands" so "first_operand", "operand_one" or "left_operand" reads better. Please take this into advisement for all the parameters in this class and the future.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+require 'rspec/autorun'
+
+class Aritmetic_operations
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: You have an entire class of repeated code here. You should remove any class code from this file and use the `require` statement to import that class into your spec file. Also, consider renaming the file to `arithmetic_operations_spec.rb` to match your class name and follow conventions.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    def self.multiply(first_number, second_number)
+        first_number * second_number
+    end
+
+    def self.divide(first_number, second_number)
+        first_number / second_number
+    end
+end
+describe Aritmetic_operations do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: `ArithmeticOperations`

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    describe "#add" do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: You chose to use class methods (the ones starting with `self` and called directly from the class). The "#method_name" format (with an octothorpe "#") is for instance methods. The equivalent for class methods is the dot format (".method_name). Please change this for every describe block. 

To be clear, this should be changed to:

```ruby
describe ".add" do
  .
  .
  .
end
```

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        it "Sum the two numbers and the result is equal to 16" do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: This line should be readable as an english language sentence. Also, the text in `it` blocks is printed to the console as the middle of a sentence (i.e. "ArithmeticOperations .add should return the sum of two numbers"), so you should use lowercase at the beginning as well. 

Finally, this description is too specific, I should be able to tell "what" the spec does without having to know the "how" details (like "and the result is equal to 16"). If I need details, I should be able to get them only by reading the code inside the block. Something like the example in the paragraph above would be better (describe behavior, not specific results).

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+             calculus = Aritmetic_operations.add(12, 4)
+            expect(calculus).to eq(16)
+        end
+    end
+
+    describe "#substraction" do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: subtraction* (please check all instances of this word in your code to correct spelling).

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        it "The first number is substracted by the second number, the result is 10" do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: Again, if you look at the console output when running this test, you'll see it doesn't make much sense as a sentence...

image

It should look like "ArithmeticOperations.subtract returns the result of subtracting two numbers" or something similar.

File: ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+    describe "#multiply" do
+        it "The result, that is 48, is the multiply of the two numbers" do
+            calculus = Aritmetic_operations.multiply(8,6)
+            expect(calculus).to eq(48)
+        end
+    end
+
+    describe "#division" do
+        it "Divide the first number by the second number and the result must be 4" do
```
Comment (HectorMg) [inline ruby-training/trainees/alan/test_katas/01/math_maddness/spec/math_maddness_spec.rb]: You should add more tests when the situation requires it... you should add a way of making sure you don't try to divide by 0 for example!! First write the spec, then make it work in your code!

**Comment by Poetickz** [comment]
Edit PR:

	> Changed the snake case to camel case.
	> Changed the name of the file to be more descriptive.
	> Modified the way to call of ArthmeticOperations class.
	> Eliminated the useless code in arithmetic_operations.rb 
	> Corrected the missing space between arguments in the calls of the methods.
	> Changed the params names in ArithmeticOperations class to be descriptive.
	> Changed the "#" to "." when is described a method in tests.
	> Changed the "it" descriptions.
	> Added division by zero error message.
	> Added test to comprobate division by zero error message.

**Pull request merged**